app/questionnaire/navigation.py

```diff
@@ -141,7 +141,7 @@ def _generate_repeated_items(self, link_names, first_location, completed_id, no_
             location.group_instance = i
             is_current_group_instance = is_current_group and i == current_group_instance
 
-            if link_names:
+            if link_names and link_names.get(i):
```

Comment:
Hmm good point, so actually maybe there's a question to be asked whether we need this fix at all. I guess the real solution here is to make sure that any answers that generate a navigation entry (repeated group) need to be sanitised/validated properly to stop that from happening?